swift/ql/lib/codeql/swift/elements/expr/MethodLookupExpr.qll

```diff
@@ -0,0 +1,24 @@
+private import codeql.swift.generated.expr.MethodLookupExpr
+private import codeql.swift.elements.expr.MethodLookupExprConstructor
+private import codeql.swift.elements.expr.Expr
+private import codeql.swift.elements.decl.Decl
+private import codeql.swift.elements.decl.MethodDecl
+private import codeql.swift.generated.Raw
+private import codeql.swift.generated.Synth
+
+class MethodLookupExpr extends Generated::MethodLookupExpr {
+  override string toString() { result = "." + this.getMember().toString() }
+
+  override Expr getImmediateBase() {
+    result = Synth::convertExprFromRaw(this.getUnderlying().getBase())
+  }
+
+  override Decl getImmediateMember() {
+    result = Synth::convertDeclFromRaw(extractDeclFromSelfApplyExpr(this.getUnderlying()))
+  }
+
+  MethodDecl getMethod() { result = this.getMember() }
```

Comment:
going with what I wrote on `DeclRefExprConstructor.qll` and [this earlier comment](https://github.com/github/codeql-c-team/issues/1362#issuecomment-1314955181), I think we should aim for (after adding `method: Expr` to `MethodLookupExpr` in `schema.py`):
```suggestion
  override Decl getImmediateMember() {
    result = this.getMethod().(DeclRefExpr).getDecl()
  }

  override Expr getImmediateMethod() { result = Synth::convertExprFromRaw(this.getUnderlying().getFunction()) }
```

Notice that this means not having to define `extractDeclFromSelfApplyExpr` at all, and relying on the hidden AST machinery in this layer in `getImmediateMember`, as `getMethod` will ignore all conversions already...